lightning/src/onion_message/packet.rs

```diff
@@ -320,13 +320,29 @@ for Payload<ParsedOnionMessageContents<<H as CustomOnionMessageHandler>::CustomM
 /// or received. Thus we read a `ControlTlvs` rather than reading a [`ForwardTlvs`] or
 /// [`ReceiveTlvs`] directly. Also useful on the encoding side to keep forward and receive TLVs in
 /// the same iterator.
+#[derive(Clone)]
 pub(crate) enum ControlTlvs {
 	/// This onion message is intended to be forwarded.
 	Forward(ForwardTlvs),
 	/// This onion message is intended to be received.
 	Receive(ReceiveTlvs),
 }
 
+impl ControlTlvs {
+	pub(crate) fn pad_to_length(mut self, length: usize) -> Self {
+		let pad_length = length.checked_sub(self.serialized_length());
+		debug_assert!(pad_length.is_some(), "Size of this packet should not be larger than the size of largest packet.");
+		let padding = Some(Padding::new(pad_length.unwrap()));
```

Comment:
I guess here we want something more like:

```rust
let padding = match length.checked_sub(self.serialized_length()) {
	Some(length) => Some(Padding::new(length)),
	None => { debug_assert!(false, "..."); None },
};
```

Otherwise, we'd panic here in release mode if there were a bug not caught by a test. Probably unlikely but better to have one less `unwrap` in our code.

Reply:
Makes sense!
Also, this structure helps to remove an extra variable creation!
Updated in [pr3177.17](https://github.com/shaavan/rust-lightning/commits/pr3177.17)
Thanks!